tests/haskell-hsc2hs-tests.el

```diff
@@ -0,0 +1,75 @@
+;; haskell-hsc2hs-tests.el --- -*- lexical-binding: t; -*-
+
+(require 'ert)
+(require 'haskell)
+(require 'haskell-test-utils)
+
+
+(defvar default-hsc "{-# LANGUAGE CPP                      #-}
+{-# LANGUAGE CApiFFI                  #-}
+{-# LANGUAGE ForeignFunctionInterface #-}
+
+module Hsc2hsTest where
+
+import           Foreign
+import           Foreign.C.String
+import           Foreign.C.Types
+
+#include <stdlib.h>
+
+newtype NUMBERS = NUMBERS { unNUMBERS :: CInt }
+    deriving (Eq,Show)
+
+#{enum NUMBERS, NUMBERS
+ , rand_max = RAND_MAX
+ }
+")
+
+(defmacro with-hsc2hs (contents &rest body)
```

Comment:
This should *always* use the `hsc2hs.sh` script. I do not want reports from people that accidentally have something strange in their paths and therefore tests accidentally pass or fail.